detection_rules/rule.py

```diff
@@ -94,10 +94,27 @@ def flattened_contents(self):
     def type(self):
         return self.contents.get('type')
 
+    @property
+    def unique_fields(self):
+        parsed = self.parsed_query
+        if parsed is not None:
+            return list(set(str(f) for f in parsed if isinstance(f, (eql.ast.Field, kql.ast.Field))))
+
     def to_eql(self):
         if self.query and self.contents['language'] == 'kuery':
             return kql.to_eql(self.query)
 
+    @classmethod
+    def get_unique_query_fields(cls, rule_contents):
+        """Get a list of unique fields used in a rule query from rule contents."""
+        query = rule_contents.get('query')
+        language = rule_contents.get('language')
+        if language in ('kuery', 'eql'):
+            parsed = kql.parse(query) if language == 'kuery' else eql.parse_query(query)
+
+            if parsed is not None:
```

Comment:
`parsed` will never be None here, so you can remove this check

Reply:
`self.parsed_query` could be none, because for non kql/eql, it just returns None instead of raising an exception